mlrun/serving/remote.py

```diff
@@ -9,21 +9,41 @@
 from urllib3.util.retry import Retry
 
 import mlrun
+from mlrun.utils import logger
 
 from .utils import (
-    StepToDict,
     _extract_input_data,
     _update_result_body,
     event_id_key,
     event_path_key,
 )
 
-http_adapter = HTTPAdapter(
-    max_retries=Retry(total=3, backoff_factor=1, status_forcelist=[500, 502, 503, 504])
-)
+default_retries = 6
+default_backoff_factor = 1
+
+
+def get_http_adapter(retries, backoff_factor):
+    if retries != 0:
+        retry = Retry(
+            total=retries or default_retries,
+            backoff_factor=backoff_factor or default_backoff_factor,
+            status_forcelist=[500, 502, 503, 504],
+            method_whitelist=[
+                "HEAD",
+                "GET",
+                "PUT",
+                "POST",
+                "DELETE",
+                "OPTIONS",
+                "TRACE",
+            ],
+        )
+    else:
+        retry = Retry(0, read=False)
```

Comment:
Why `read=False`?
```
    :param int read:
        How many times to retry on read errors.

        These errors are raised after the request was sent to the server, so the
        request may have side-effects.

        Set to ``0`` to fail on the first retry of this type.
```

Reply:
looks from the code that False is a valid value, i took this way from the `HTTPAdapter()`  code